demonstrations/qsim_supremacy.py

```diff
@@ -0,0 +1,567 @@
+# correspond to all horizontally neighbouring qubits (in a row), and C and
+# D to the vertically neighbouring qubits (in a column). This is depicted
+# in the figure below, where you can also see how the single-qubit gates
+# are applied, as well as the cycles, each consisting of a layer of
+# single-qubit gates and a pair of two-qubit gates. Note that each coloured
+# two-qubit gate represented in the image is implemented as the two
+# consecutive gates iSWAP and CPhase in this demo.
+#
+# .. figure:: ../demonstrations/qsim_supremacy/supremacy_circuit.png
+#     :align: center
+#     :width: 90%
+#
+#     **Image taken from Arute, F., Arya, K., Babbush, R. et al.** [#Arute2019]_
+#
+# The logic below iterates through all connections and returns a
+# dictionary, ``gate_order``, where the keys are the connection labels
+# between different qubits and the values are lists of all neighbouring
+# qubit pairs. We will use this dictionary inside the circuit to iterate
+# through the different pairs and apply the two two-qubit gates that we
+# just defined above. The way we iterate through the dictionary will depend
+# on a gate sequence defined in the next section.
+#
+
+from itertools import combinations
+
+gate_order = {"A":[], "B":[], "C":[], "D":[]}
+for i, j in combinations(qubits, 2):
+    wire_1 = qb2wire[i]
+    wire_2 = qb2wire[j]
+    if i in j.neighbors():
+        if i.row == j.row and i.col % 2 == 0:
+            gate_order["A"].append((wire_1, wire_2))
+        elif i.row == j.row and j.col % 2 == 0:
+            gate_order["B"].append((wire_1, wire_2))
+        elif i.col == j.col and i.row % 2 == 0:
+            gate_order["C"].append((wire_1, wire_2))
+        elif i.col == j.col and j.row % 2 == 0:
+            gate_order["D"].append((wire_1, wire_2))
+
+
+######################################################################
+# At this point we can define the gate sequence, which is the order the
+# two-qubit gates are applied to the different qubit pairs. For example,
+# ``["A", "B"]`` would mean that the two-qubit gates are first applied to
+# all qubits connected with label A, and then, during the next full cycle,
+# the two-qubit gates are applied to all qubits connected with label B.
+# This would then correspond to a 2-cycle run (or a circuit with a depth of
+# 2).
+#
+# While we can define any patterns we'd like, the two gate sequences below
+# are the ones that are used in the supremacy paper. The shorter one is
+# used for their classically verifiable benchmarking. The slightly
+# longer sequence, which is much harder to simulate classically, is used
+# for estimating the cross-entropy fidelity in what they call the "supremacy
+# regime". We will use the shorter gate sequence for the following
+# demonstration; feel free to play around with other combinations.
+#
+
+m = 14  # number of cycles
+
+# gate_sequence = np.resize(["A", "B", "C", "D", "C", "D", "A", "B"], m)
+gate_sequence = np.resize(["A", "B", "C", "D"], m)
```

Comment:
It might look slighter nice to leave both uncommented, e.g.,
```suggestion
gate_sequence_longer = np.resize(["A", "B", "C", "D", "C", "D", "A", "B"], m)
gate_sequence = np.resize(["A", "B", "C", "D"], m)
```

Reply:
(even though `gate_sequence_longer` isn't used)